src-tauri/src/global/algo_store.rs

```diff
@@ -0,0 +1,107 @@
+#![allow(dead_code)]
+
+use nalgebra::DMatrix;
+use std::collections::HashMap;
+
+use crate::aux_data_structures::stoer_wagner_ds::StoerWagnerGraph;
+use crate::graph::{edge::Edge, graph_ds::Graph};
+use petgraph::graph::NodeIndex;
+
+use crate::algorithms::articulation_point::articulation_point;
+use crate::algorithms::diffcen::diffusion_centrality;
+use crate::algorithms::stoer_wagner::{recover_mincut, stoer_wagner};
+
+pub struct AlgoStore {
+    pub aps: Option<Vec<petgraph::graph::NodeIndex>>,
+    pub mincut: Option<Vec<Edge>>,
+    pub diff_cent: Option<HashMap<String, HashMap<String, f64>>>,
+    pub most_sim_t: Option<Graph>,
+    pub pred_state: Option<Graph>,
+}
+
+impl AlgoStore {
```

Comment:
Does it make sense to store algo results here? Not sure. We can also save it in global state or create another struct.